src/app/components/data-field-edit-dialog/data-field-edit-dialog.component.ts

```diff
@@ -91,7 +86,7 @@ export class DataFieldEditDialogComponent {
 
   addDataOption(dataField: DataField) {
     this.addOrEditDataOption({
-      displayOrder: this.data.dataFieldOptions.length + 1,
+      displayOrder: this.data.dataField.dataOptions.length + 1,
```

Comment:
We should probably make it one greater than the current greatest value.

Reply:
This is just the value that gets pre-populated into the edit modal.  The user can correct it if it needs adjusting.  We currently do nothing to reorder DataOptions automatically.  Go ahead and add a Jira ticket to add that functionality.